content/tutorials/trainapp-fitbit.textile

```diff
@@ -0,0 +1,550 @@
+let latitude, longitude, timeToArrival, walkingTime;
+
+function locationSuccess(position) {
+  latitude = position.coords.latitude;
+  longitude = position.coords.longitude;
+  if (JSON.parse(settingsStorage.getItem("line")) === null) return;
+  // Reading the origin station from Companion Settings
+  const station = JSON.parse(settingsStorage.getItem("origin")).name; 
+  if (station === "") return;
+
+  // Replace the space with plus so that it can be passed to Google API  
+  let dest = station.split(" ").join("+");  
+  let googleUrl = `https://maps.googleapis.com/maps/api/distancematrix/json?
+  origins=${latitude},${longitude}&destinations=${dest}&mode=walking&key=${apiKey}`;
+  fetch(googleUrl, {
+    method: "GET"
+  })
+    .then(function(res) {
+      return res.json();
+    })
+    .then(function(data) {
+      let myData = data;
+      // Convert the time to mins
+      walkingTime = Math.floor(myData.rows[0].elements[0].duration.value / 60); 
+    })
+    .catch(err => console.log("[FETCH]: " + err));
+}
+
+function locationError(error) {
+  console.log("Error: " + error.code, "Message: " + error.message);
+}
+```
+"See this step in Github":https://github.com/ably/tutorials/commit/de04d1daa6d2440163513de7efdb9b3e4a79f55a
+
+h2(#send-message). Step 6 - Subscribe to Ably TFL Hub & send the message to app
+
+Now using the pub/sub mechanism of Ably, we subscribe to the @Ably TFL Hub@ with the selected line & stationId (station & towards) .
+The new data to the hub is published automatically for the given input and we just subscribe to the channel to recieve that.
+We get the @walkingTime@ using the Google API to which we send the @current location@ & @destination(station)@ .
+Then using the data that the Ably TFL Hub provides us we get the list of upcoming trains and calculate then the arrival time of the next train.
+Now using the @messaging@ module of Fitbit we send the data from @companion(here)@ to the @app@ .
+The data the consists of three times - walking time, time of next train & leave time(that the person uses to decide when to leave the current location to catch the train) . 
+Add the following code in your @index.js@ file which is inside the @companion@ folder: 
+
+```[javascript]
+function sendMessage() {
+ // Get the line, origin station & towards from Companion settings through settingsStorage
+   const line = JSON.parse(settingsStorage.getItem("line")).values[0].name;
+   const station = JSON.parse(settingsStorage.getItem("origin")).name;
+   const towards = JSON.parse(settingsStorage.getItem("towards")).name;
+ 
+   if (station === "" || towards === "") return
+     // Get stationId for the selected line and origin station,
+     // using the StationsIDMapping static data
+     const stationId = StationsIDMapping.filter(function(train) {
+       return train.line === line && train.station === station;
+     });
+
+//We use the Ably TFL Hub, to recieve update for the line & stationid (station & towards)   
```

Comment:
Tiny point, space between `//` and `We` would be nice for consistency with other comments.

Reply:
`recieve` => `receive`